src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1887,6 +1938,27 @@ def update_nat_gateway_profile(self, mc: ManagedCluster) -> ManagedCluster:
             )
         return mc
 
+    def update_pod_identity_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update pod identity profile for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        if self.context.get_enable_pod_identity():
+            if not _is_pod_identity_addon_enabled(mc):
+                # we only rebuild the pod identity profile if it's disabled before
+                _update_addon_pod_identity(
+                    mc,
+                    enable=True,
+                    allow_kubenet_consent=self.context.get_enable_pod_identity_with_kubenet(),
+                    models=self.models.pod_identity_models
+                )
+
+        if self.context.get_disable_pod_identity():
```

Comment:
nit: use `elseif` here? Since we don't allow to enable & disable at the same time

Reply:
also I think we can move the `_ensure_mc` call into the branch to save some repeated calls

Reply:
Yeah, for other pairs of `enable_xxx` and `disable_xxx` parameters, there is usually a mutual exclusion check on the cli side. Here I just moved the original implementation, if necessary, I will add a check in the getter functions.

Reply:
Updated in commit [#059868e](https://github.com/Azure/azure-cli-extensions/pull/4243/commits/059868ed2acd69dfd246b17785c35a4630f21ff1).